lib/index.js

```diff
@@ -512,114 +489,155 @@ internals.diffColumns = (migrationGroup, joinTables, mode, cb) => {
             const tmpColumnCompiler = knex.client.columnCompiler(tmpTableBuilder, tmpColumnBuilder);
             const columnType = tmpColumnCompiler.getColumnType();
 
-            // columnType.split('(')[0] is specific for varchar values. Knex appends (255) to this columnType
-            // if the varchar length can go up to 255
-            // Remove the output (255) specified lengths from varchar types.
-            // At this time we only care about the column types
-            collector[columnName] = columnType.split('(')[0]; // To the right of this paren is specified the length if available (e.g. (255))
+            // columnType.split('(')[0] is specific for varchar values. Knex appends (255) (or whichever length) to this columnType
+            // Remove the output (xyz) specified lengths from varchar types.
+            // At this time we only care about the column types not the lengths
+            collector[columnName] = columnType.split('(')[0];
             return collector;
         }, {});
 
-        internals.getColumnsForTable(knex, modelName)
-            .then((dbSchema) => {
+        internals.getColumnsForTable(knex, modelName, (err, dbSchema) => {
 
-                //////////// internals.convertDb2KnexSchema()
+            if (err) {
+                return next(err);
+            }
 
-                const dbColsToConvert = Object.keys(dbSchema)
-                    .filter((colName) => Mappings.ignoreColumns.indexOf(dbSchema[colName]) === -1);
+            const dbColsToConvert = Object.keys(dbSchema)
+                .filter((colName) => {
 
-                const db2ColumnCompilerSchema = dbColsToConvert.reduce((collector, columnName) => {
+                    const colType = dbSchema[colName];
+                    const [err, val] = Mappings.convertFuncs.getAliasVal(colType);
+                    if (!!err) {
+                        internals.skippedCols.push(`model: ${modelName}, colName: ${colName}, colType: ${colType}`);
+                        return false;
+                    }
+                    return true;
```

Comment:
Totally your call but you can always have a single return statement: `return !err`.